differences across the product with respect to customer characteristics.

##### Univariate 
1 Age range for each product.  
2 Marital Status wise classification of products.   
3 Gender wise classification for each product .  
4 Product wise Average fitness rank of customers.   
5 Usage of each product per week .   
6 Income range of customers for each product .   
7 is education a factor between customers of each product .   
8 Average Miles per week for each product .    


##### Bivariate.   
Relation between age and fitness rank.  
Relation between age and income.  
How usage varies with different genders.  
How usage varies with Marital Status.   
Relation with fitness and miles.    
Relation between education and usage,miles



#### Importing Libraries

In [21]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 
import warnings
warnings.filterwarnings("ignore")

#### Loading data

In [5]:
df = pd.read_csv('datasets/aerofit_treadmill_data.csv')
df.sample(10)

Product  Age  Gender  Education MaritalStatus  Usage  Fitness  Income  \
86    KP481   21    Male         12     Partnered      2        2   32973   
74    KP281   41    Male         16     Partnered      4        3   54576   
174   KP781   38    Male         18     Partnered      5        5  104581   
130   KP481   35  Female         16        Single      3        2   50028   
100   KP481   25  Female         14     Partnered      5        3   47754   
44    KP281   28  Female         14     Partnered      2        3   46617   
85    KP481   21    Male         16     Partnered      2        2   34110   
102   KP481   25  Female         14        Single      2        3   43206   
141   KP781   22    Male         16        Single      3        5   54781   
23    KP281   24  Female         16     Partnered      5        5   44343   

     Miles  
86      53  
74     103  
174    150  
130     64  
100    106  
44      56  
85      42  
102     64  
141    120  
23     188

## Data Exploration and Processing

In [7]:
df.shape

(180, 9)

In [22]:
df.columns

Index(['Product', 'Age', 'Gender', 'Education', 'MaritalStatus', 'Usage',
       'Fitness', 'Income', 'Miles'],
      dtype='object')

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Product        180 non-null    object
 1   Age            180 non-null    int64 
 2   Gender         180 non-null    object
 3   Education      180 non-null    int64 
 4   MaritalStatus  180 non-null    object
 5   Usage          180 non-null    int64 
 6   Fitness        180 non-null    int64 
 7   Income         180 non-null    int64 
 8   Miles          180 non-null    int64 
dtypes: int64(6), object(3)
memory usage: 12.8+ KB


##### The DataFrame has columns of datatype `object`. Since each such column only includes predefined labels, so we can convert the datatype to `category`.  

#### Type Conversion

In [27]:
df['Product'] = df['Product'].astype('category')
df['Gender'] = df['Gender'].astype('category')
df['MaritalStatus'] = df['MaritalStatus'].astype('category')

In [30]:
df.dtypes

Product          category
Age                 int64
Gender           category
Education           int64
MaritalStatus    category
Usage               int64
Fitness             int64
Income              int64
Miles               int64
dtype: object

In [38]:
# Detecting Duplicates 

df.duplicated().sum() # No duplicate values in the dataset

0

In [41]:
# Detecting Null Values

df.isna().sum() # No missing value detected in the dataset

Product          0
Age              0
Gender           0
Education        0
MaritalStatus    0
Usage            0
Fitness          0
Income           0
Miles            0
dtype: int64

### Statistical Summary

In [34]:
df.describe(include='all')

Product         Age Gender   Education MaritalStatus       Usage  \
count      180  180.000000    180  180.000000           180  180.000000   
unique       3         NaN      2         NaN             2         NaN   
top      KP281         NaN   Male         NaN     Partnered         NaN   
freq        80         NaN    104         NaN           107         NaN   
mean       NaN   28.788889    NaN   15.572222           NaN    3.455556   
std        NaN    6.943498    NaN    1.617055           NaN    1.084797   
min        NaN   18.000000    NaN   12.000000           NaN    2.000000   
25%        NaN   24.000000    NaN   14.000000           NaN    3.000000   
50%        NaN   26.000000    NaN   16.000000           NaN    3.000000   
75%        NaN   33.000000    NaN   16.000000           NaN    4.000000   
max        NaN   50.000000    NaN   21.000000           NaN    7.000000   

           Fitness         Income       Miles  
count   180.000000     180.000000  180.000000  
unique         NaN            NaN         NaN  
top            NaN            NaN         NaN  
freq           NaN            NaN         NaN  
mean      3.311111   53719.577778  103.194444  
std       0.958869   16506.684226   51.863605  
min       1.000000   29562.000000   21.000000  
25%       3.000000   44058.750000   66.000000  
50%       3.000000   50596.500000   94.000000  
75%       4.000000   58668.000000  114.750000  
max       5.000000  104581.000000  360.000000

### Observations

    • There is no missing and duplicate values in data set
    • Dataset contains 3 unique products
    • KP281 is the most frequent product 
    • Out of 180 data points, 104's gender is male rest are the females
    • Out of 180 data points, 107's Marital Status is Partnered rest are the singles
    • Age varies between 18<=Age<=50. Out of which 75% of persons having age <= 33
    

## Non-Graphical Analysis

In [43]:
df['Product'].unique()

['KP281', 'KP481', 'KP781']
Categories (3, object): ['KP281', 'KP481', 'KP781']

In [44]:
df['Product'].value_counts()

Product
KP281    80
KP481    60
KP781    40
Name: count, dtype: int64

In [45]:
df['Gender'].unique()

['Male', 'Female']
Categories (2, object): ['Female', 'Male']

In [46]:
df['Gender'].value_counts()

Gender
Male      104
Female     76
Name: count, dtype: int64

In [47]:
df['MaritalStatus'].unique()

['Single', 'Partnered']
Categories (2, object): ['Partnered', 'Single']

In [48]:
df['MaritalStatus'].value_counts()

MaritalStatus
Partnered    107
Single        73
Name: count, dtype: int64

### Unique values in numerical columns

In [53]:
df.nunique()

Product           3
Age              32
Gender            2
Education         8
MaritalStatus     2
Usage             6
Fitness           5
Income           62
Miles            37
dtype: int64

In [54]:
df['Age'].unique()

array([18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34,
       35, 36, 37, 38, 39, 40, 41, 43, 44, 46, 47, 50, 45, 48, 42])

In [55]:
df['Education'].unique()

array([14, 15, 12, 13, 16, 18, 20, 21])

In [56]:
df['Usage'].unique()

array([3, 2, 4, 5, 6, 7])

In [57]:
df['Fitness'].unique()

array([4, 3, 2, 1, 5])

In [61]:
df['Income'].unique()

array([ 29562,  31836,  30699,  32973,  35247,  37521,  36384,  38658,
        40932,  34110,  39795,  42069,  44343,  45480,  46617,  48891,
        53439,  43206,  52302,  51165,  50028,  54576,  68220,  55713,
        60261,  67083,  56850,  59124,  61398,  57987,  64809,  47754,
        65220,  62535,  48658,  54781,  48556,  58516,  53536,  61006,
        57271,  52291,  49801,  62251,  64741,  70966,  75946,  74701,
        69721,  83416,  88396,  90886,  92131,  77191,  52290,  85906,
       103336,  99601,  89641,  95866, 104581,  95508])

In [62]:
df['Miles'].unique()

array([112,  75,  66,  85,  47, 141, 103,  94, 113,  38, 188,  56, 132,
       169,  64,  53, 106,  95, 212,  42, 127,  74, 170,  21, 120, 200,
       140, 100,  80, 160, 180, 240, 150, 300, 280, 260, 360])

## Graphical Analysis